HomeAutomationServer/Models/AppCache.cs

```diff
@@ -46,51 +51,56 @@ static public bool AddDeviceBlob_DEBUG(JObject blob)
 
 		/*************** END DEBUG MODE METHOD *************/
 
-        static public bool AddDeviceBlob(JObject blob)
-        {
-            deviceBlobs[(string)blob["deviceID"]] = blob;
-
-            if (deviceBlobs[(string)blob["deviceID"]] == blob)
-            {
-                try
-                {
-                    AmazonSimpleNotificationServiceClient snsClient = new AmazonSimpleNotificationServiceClient("AKIAJM2E3LGZHJYGFSQQ", "p3Qi8DAXj+XHAH+ny7HrlRyleBs5V5DJv77zKK3T", Amazon.RegionEndpoint.USEast1);
-                    snsClient.Publish("arn:aws:sns:us-east-1:336632281456:MyTopic", "New Device Updates");
-                }
-
-                catch (Exception ex)
-                {
-                    LogFile.AddLog("AppCache -- Could not send Push Notification: " + ex.Message + "\n");
-                    return false;
-                }
-
-                return true;
-            }
-
-            return false;
-        }
-
-        static public JToken GetDeviceBlob(string deviceID)
-        {
-            JToken blob = deviceBlobs[deviceID];
-            deviceBlobs[deviceID].Remove();
-            return blob;
-        }
-
-        static public JArray GetAllBlobs()
-        {
-            JArray blobs = deviceBlobs;
-            deviceBlobs.RemoveAll();
-            return blobs;
-        }
-
-        static public int GetBlobCount()
-        {
-            return deviceBlobs.Count;
-        }
-
-        ////////////////////////////////////////////////////////////////////////////////////////
-        //
-        // other blob methods
-    }
+		static public bool AddDeviceBlob(JObject blob)
+		{
+			string blob_string = blob.ToString ();
+			Device dev = Interfaces.DeserializeDevice(blob_string, null, null, new TimeFrame()); // to convert the JSON blob to an actual Device object.
+			string devID = (string)blob["deviceID"];
+
+			deviceBlobs.Add(devID, dev);
+
+			if (deviceBlobs.Contains(dev)) {
+				try
+				{
+					AmazonSimpleNotificationServiceClient snsClient = new AmazonSimpleNotificationServiceClient("AKIAJM2E3LGZHJYGFSQQ", "p3Qi8DAXj+XHAH+ny7HrlRyleBs5V5DJv77zKK3T", Amazon.RegionEndpoint.USEast1);
```

Comment:
This was the same key that was exposed in the original breach which was revoked long ago. So there's nothing to worry about here. However, all references to old keypairs should probably be removed to prevent false alarms.